value.go

```diff
@@ -317,7 +350,43 @@ type valueSimpleObject struct {
 	valueObjectBase
 	upValues bindingFrame
 	fields   valueSimpleObjectFieldMap
-	asserts  []ast.Node
+	asserts  []unboundField
+}
+
+func checkAssertionsHelper(e *evaluator, obj valueObject, curr valueObject, superDepth int) error {
+	switch curr := curr.(type) {
+	case *valueExtendedObject:
+		err := checkAssertionsHelper(e, obj, curr.right, superDepth)
+		if err != nil {
+			return err
+		}
+		err = checkAssertionsHelper(e, obj, curr.left, superDepth+curr.right.inheritanceSize())
```

Comment:
spaces around +

Reply:
also this

Reply:
This is how gofmt formats it:
https://github.com/golang/go/issues/1206